### PR TITLE
tor: disabled tests on aarch32

### PR DESCRIPTION
**Pull request by dasj19**
## Description of changes

I added aarch32 on the "doNotCheck" list because the current upstream tests are failing for aarch32 builds.

Removed aarch64 darwin from `doCheck` as requested.

## Things done

Built on a x86_64 remote for an aarch32 machine. The build succeeds with the current doCheck expression.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Ping maintainers: @thoughtpolice @joachifm @prusnak

### CODE REVIEW

File: pkgs/tools/security/tor/default.nix [hunk 1]
```diff
@@ -77,7 +77,30 @@ stdenv.mkDerivation rec {
...
+  #   FAIL src/test/test_sandbox.c:190: assert(rc OP_EQ -1): 0 vs -1
+  #   [chmod_filename FAILED]
+  # sandbox/chown_filename: [forking] Feb 06 17:43:14.561 [err] install_syscall_filter(): Bug: (Sandbox) failed to load: -125 (Operation canceled)! Are you sure that your kernel has seccomp2 support? The sandbox won't work without it. (on Tor 0.4.8.10 )
+  #   FAIL src/test/test_sandbox.c:208: assert(rc OP_EQ -1): 0 vs -1
+  #   [chown_filename FAILED]
+  # sandbox/rename_filename: [forking] Feb 06 17:43:14.629 [err] install_syscall_filter(): Bug: (Sandbox) failed to load: -125 (Operation canceled)! Are you sure that your kernel has seccomp2 support? The sandbox won't work without it. (on Tor 0.4.8.10 )
+  #   FAIL src/test/test_sandbox.c:228: assert(rc OP_EQ -1): 0 vs -1
+  #   [rename_filename FAILED]
+
+  doCheck = !(stdenv.isDarwin && stdenv.isAarch64) && !stdenv.isAarch32;
```
Comment (prusnak) [inline pkgs/tools/security/tor/default.nix:103]: I assume you tried aarch32 on Linux. I suggest this change:

```suggestion
  doCheck = !(stdenv.isDarwin && stdenv.isAarch64) && !(stdenv.isLinux && stdenv.isAarch32);
```

Also, does the resulting binary work? Is it just the tests which do fail?

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/tor/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/tools/security/tor/default.nix:103]: > do you think I should also report upstream that the test fails under Darwin Aarch64 and Linux Aarch32 

yes, please, that would be perfect - two different issues and we can link them both

Comment (dasj19) [reply pkgs/tools/security/tor/default.nix]: super, thanks, I just pushed your suggestion and requested a gitlab account at the tor project.

**Comment by dasj19** [comment]
I submitted upstream that the tests are failing on linux/aarch32 https://gitlab.torproject.org/tpo/core/tor/-/issues/40912
@prusnak Can you recheck the darwin/aarch64 build without `doCheck` disabled? I want to make sure those tests are still failing. And If they still do I can submit upstream a bug for them. Thanks.

**Comment by prusnak** [comment]
> Can you recheck the darwin/aarch64 build without `doCheck` disabled?

I just tried it and darwin/aarch64 issues are gone. You can remove the related comment and use only:

```
doCheck = !(stdenv.isLinux && stdenv.isAarch32);
```

There is another issue (torsocks being broken on darwin), but I will address that in another PR.

**Comment by dasj19** [comment]
@prusnak thanks for checking and feedback. I added a new commit that deals with removing the darwin/aarch64 rule from `doCheck`.

**Review by prusnak — APPROVED** [review]

**Pull request merged**